src/realm/object-store/c_api/http.cpp

```diff
@@ -70,6 +72,8 @@ class CNetworkTransport final : public GenericNetworkTransport {
                                        c_headers.size(),
                                        request.body.data(),
                                        request.body.size()};
+        auto completion_data = std::unique_ptr<HttpCompletionData>(
```

Comment:
Unfortunately, I haven't been able to get `std::make_unique` to work (that was what I tried initially), since the compiler complains that there isn't a constructor that takes two arguments and I can't use (or figure out how to use) aggregate initialization here.

Reply:
you just need to actually make a constructor for HttpCompletionData.